packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -438,11 +438,21 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     }
   }, [connectionState, pushToast])
 
-  // Reset `focusPath` when `documentId` or `params.path` changes
   useEffect(() => {
-    // Reset focus path
-    setFocusPath(params.path ? pathFromString(params.path) : [])
-  }, [documentId, params.path])
+    // The path that is specified in the URL
+    const paramsPath = params.path ? pathFromString(params.path) : []
+
+    // The first field in the schema
+    // TODO: consider adding logic to check if the first field is enabled
+    const firstField = schemaType?.fields?.[0].name
```

Comment:
I think we could do something like this to make sure that the field is not hidden. The `members` array do not contain any fields that are hidden.

```ts
const firstMemberFieldName = (formStateRef.current?.members?.[0] as FieldMember)?.name
```

Reply:
I'm still not convinced that we should set the focus element in `DocumentPaneProvider.tsx` instead of in the `FormView.tsx`. 
In your suggestion here: if the first member does not have a `name` field, i.e. if it instead has objects, it will be `undefined`. I think it will be cleaner to use `focusFirstDescendent` to avoid covering edge cases. Even if this will affect the presence location

Reply:
`focusFirstDescendant` is currently used in `Dialog` as well, so using in the `FormView` is more consistent